ksqldb-parser/src/main/java/io/confluent/ksql/schema/ksql/DefaultSqlValueCoercer.java

```diff
@@ -37,28 +37,33 @@
 
   INSTANCE;
 
-  private static final Map<SqlBaseType, BiFunction<Number, SqlType, Optional<Number>>> UPCASTER =
-      ImmutableMap.<SqlBaseType, BiFunction<Number, SqlType, Optional<Number>>>builder()
-          .put(SqlBaseType.INTEGER, (num, type) -> Optional.of(num.intValue()))
-          .put(SqlBaseType.BIGINT, (num, type) -> Optional.of(num.longValue()))
-          .put(SqlBaseType.DOUBLE, (num, type) -> Optional.of(num.doubleValue()))
+  private static final Map<SqlBaseType, BiFunction<Number, SqlType, Result>> UPCASTER =
+      ImmutableMap.<SqlBaseType, BiFunction<Number, SqlType, Result>>builder()
+          .put(SqlBaseType.INTEGER, (num, type) -> Result.of(num.intValue()))
+          .put(SqlBaseType.BIGINT, (num, type) -> Result.of(num.longValue()))
+          .put(SqlBaseType.DOUBLE, (num, type) -> Result.of(num.doubleValue()))
           .put(SqlBaseType.DECIMAL, (num, type) -> {
             try {
-              return Optional.ofNullable(
+              return Result.of(
                   DecimalUtil.ensureFit(
                       new BigDecimal(String.format("%s", num)),
                       (SqlDecimal) type));
             } catch (final Exception e) {
-              return Optional.empty();
+              return Result.failure();
             }
           }).build();
 
   @Override
-  public Optional<?> coerce(final Object value, final SqlType targetType) {
+  public Result coerce(final Object value, final SqlType targetType) {
```

Comment:
refactored to allow `coerce` to return a _successful_ coercion with a `no value` result, i.e. coercion of nulls.  Previous return value would not differentiate between a failure and 'no value'